### PR TITLE
Add macro for zeroex_*_api_fills

### PR DESCRIPTION
**Pull request by Hosuke**
cc. @RantumBits for review.

### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: dbt_subprojects/dex/macros/models/_project/zeroex/zeroex_api_fills.sql [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        )
+        THEN (all_tx.maker_token_amount_raw / pow(10, mp.decimals)) * mp.price
+        WHEN taker_token IN (
+            {% for address in stablecoin_addresses %}
+                {{ "0x" + address[2:] }}{% if not loop.last %},{% endif %}
+            {% endfor %}
+        )
+        THEN (all_tx.taker_token_amount_raw / pow(10, tp.decimals)) * tp.price
+        ELSE COALESCE((all_tx.maker_token_amount_raw / pow(10, mp.decimals)) * mp.price, (all_tx.taker_token_amount_raw / pow(10, tp.decimals)) * tp.price)
+    END AS volume_usd,
```
Comment (Hosuke) [inline dbt_subprojects/dex/macros/models/_project/zeroex/zeroex_api_fills.sql:112]: This part may be too complicated, we may use trusted tokens in future.

**Comment by Hosuke** [comment]
The [test failure](https://github.com/duneanalytics/spellbook/actions/runs/10358652382/job/28673331641?pr=6546#step:12:33) is due to maker address discrepency in the seed:


test_description | result_model | expected_seed | equality_check | tx_hash | evt_index
-- | -- | -- | -- | -- | --
equality test: maker | 0x00000012556e6973776170563300000000000000 | 0x000012556e697377617056330000000000000000 | false | 0x2c127cc4ff2951197c84a7752bde56d9b3ce9b65d180d4a7ca73465702844038 | 9
equality test: maker | 0x0000001942616c616e6365725632000000000000 | 0x00001942616c616e636572563200000000000000 | false | 0x2c127cc4ff2951197c84a7752bde56d9b3ce9b65d180d4a7ca73465702844038 | 15
equality test: maker | 0x0000001942616c616e6365725632000000000000 | 0x00001942616c616e636572563200000000000000 | false | 0xec22d7c79f9276dddebb8c94b15313f1602ae9c964815ffaa39800e803e10488 | 9
equality test: maker | 0x0000000253757368695377617000000000000000 | 0x0000025375736869537761700000000000000000 | false | 0x95db7724a6ff1272121cf7eee68bbc5b9f177d59c790e9cb130bc670fff22d33 | 7
equality test: maker | 0x00000012556e6973776170563300000000000000 | 0x000012556e697377617056330000000000000000 | false | 0x95db7724a6ff1272121cf7eee68bbc5b9f177d59c790e9cb130bc670fff22d33 | 12

Example log: https://arbiscan.io/tx/0x2c127cc4ff2951197c84a7752bde56d9b3ce9b65d180d4a7ca73465702844038#eventlog

We may either fix the seed record or shift maker address parsing.

**Comment by Hosuke** [comment]
While I acknowledge that my current implementation may not be entirely accurate, my intention is to provide a conceptual example demonstrating how we could potentially apply this macro to zeroex spells. This serves as a starting point for further discussion and refinement.

**Comment by jeff-dude** [comment]
closing for now, we will keep branch to revisit as needed

**Pull request closed without merging**